auth/credentials/oauth.go

```diff
@@ -31,11 +39,33 @@ type OAuthTokenSource struct {
 	clientSecret string
 	userAgent    string
 	tokenURL     string
+	revokeURL    string
 	token        *Token
 	tokenCache   LocalTokenCache
 	ctx          context.Context
 }
 
+func (c *OAuthTokenSource) RevokeToken() error {
+	tokenString, err := c.tokenCache.RetrieveToken(c.ctx)
+	if err != nil {
+		return err
+	}
+	if tokenString != nil && *tokenString != "" {
+		err := c.revokeTokenInRemoteServer(*tokenString)
+		if err != nil {
+			return err
+		}
+
+		// Remove token from cache
```

Comment:
nit: i normally prefer code comments about WHY instead of WHAT we're doing. for example in this case we might want to have a tokenCache.RemoveToken or something similar:

```

func (s *MyTokenCache) RemoveToken(ctx context.Context) error {
   return s.SaveToken(ctx, "")
}
```

Reply:
> WHY instead of WHAT 
```suggestion
		// Revoked token can be removed from cache
```

>  for example in this case we might want to have a tokenCache.RemoveToken

I think this is a separate suggestion to extend the interface. 
I'm going to move from `*string` to `string` which will make interface more clear and less prone to confusion. 
Remove will be redundant in this case.